dht/i_partitioner.cc

```diff
@@ -295,43 +295,43 @@ bool ring_position::less_compare(const schema& s, const ring_position& other) co
     return tri_compare(s, other) < 0;
 }
 
-int ring_position_tri_compare(const schema& s, ring_position_view lh, ring_position_view rh) {
+std::strong_ordering ring_position_tri_compare(const schema& s, ring_position_view lh, ring_position_view rh) {
     auto token_cmp = tri_compare(*lh._token, *rh._token);
     if (token_cmp != 0) {
-        return token_cmp < 0 ? -1 : 1;
+        return token_cmp;
     }
     if (lh._key && rh._key) {
         auto c = lh._key->legacy_tri_compare(s, *rh._key);
         if (c) {
```

Comment:
In a previous patch you fixed if's like if(!c) to be if(c==0) to make it clear this is not a boolean but a strong ordering. Isn't it prudent to do the same with "if(c)" here, making it "if(c != 0)" to make it clearer this is not a boolean?

Reply:
This will get fixed when I patch legacy_tri_compare() to return std::strong_ordering. I'd rather not convert the entire code base at the same time.